fix(formatter/sort_imports): Handle internal prefixes correctly

leaysgur opened the pull request:
| |Before|After|
|-|-|-|
| `from "@scoped/foo"`| internal (🐛) | external (✨) |
| `from "@/foo"` | internal | internal |

ghiscoding commented on crates/oxc_formatter/src/ir_transform/sort_imports/compute_metadata.rs:
```diff
@@ -365,7 +365,7 @@ fn to_path_kind(source: &str) -> ImportPathKind {
 
     // TODO: This can be changed via `options.internalPattern`
-    if source.starts_with('~') || source.starts_with('@') {
+    if source.starts_with("~/") || source.starts_with("@/") {
```
great, this now matches Perfectionist defaults 👍🏻 
https://perfectionist.dev/rules/sort-imports#internalpattern

leaysgur committed:
fix(formatter/sort_imports): Handle internal prefixes correctly (#16128)

| |Before|After|
|-|-|-|
| `from "@scoped/foo"`| internal (🐛) | external (✨) |
| `from "@/foo"` | internal | internal |
